Update from `cisagov/skeleton-python-library`

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request brings in upstream changes from [cisagov/skeleton-python-library] to bring this repository into our current development and testing practices.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Currently the GitHub Actions testing is not running in #5 and this is likely because of how out of date the configuration is in this project. Additionally if we are making changes to this project it should be brought up to date and we should ensure it will keep getting automatically updated with [cisagov/action-lineage] until such a time as we archive the project.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I coordinated with @king-alexander and he verified functionality with a branch using #5 rebased on this branch. He manually built a new Docker image, verified functionality, then built a new Docker image again this time with an empty `INSTALL_IPYTHON` to ensure updated logic worked with the other Dockerfile changes.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

[cisagov/skeleton-python-library]: https://github.com/cisagov/skeleton-python-library
[cisagov/action-lineage]: https://github.com/cisagov/action-lineage

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #56 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

jsf9k committed:
Insist that the cisagov devs are the owners of the .github directory

This additional clause must remain at the _end_ of the CODEOWNERS file
so that it cannot be overridden by a later clause.

We want to make it so that all the .github files including CODEOWNERS
are protected so only code owners (the dev team) can approve
modifications to them.

This will prevent configuration changes from breaking Actions and
other management-type functions that the files in this directory
control. By setting the .github files/folder to require code owner
approval for changes, workflow and management changes will require dev
team review and checking.

Resolves #56.

jsf9k committed:
Remove offending slash

Thanks to @dav3r and @mcdonnnj for the suggestion.

jsf9k committed:
Use the python version output by actions/setup-python

There is no need to run python code to determine the python version.
Resolves #58.

See here for details:
https://github.com/actions/setup-python/blob/main/action.yml#L14-L16

mcdonnnj committed:
Make sure the requirements in setup.py are sorted

Sort the install_requires and extras_require["test"] arguments by key.

mcdonnnj committed:
Organize pre-commit hooks

Group related pre-commit hooks together. Make sure that hooks are
alphabetically sorted within those groups.

mcdonnnj committed:
Change how the version number is extracted in setup.py

Switch to the file line reading version of extracting the version from
https://packaging.python.org/guides/single-sourcing-package-version/
instead of the exec method on the same page. The exec method required us to use
a "# nosec" to manually disable Bandit checking on that line. Although that
method is more straightforward, I do not feel that it is worth using an exec in
the codebase when another option is available.

mcdonnnj committed:
Add comment explaining a Flake8 check disable

mcdonnnj committed:
Resolve LGTM alert

mcdonnnj committed:
Merge pull request #60 from cisagov/organize_pre-commit_requirements

Organize and Sort pre-commit Hooks

jsf9k committed:
Merge branch 'develop' into improvement/insist-that-devs-own-gihub-dir

jsf9k committed:
Merge pull request #57 from cisagov/improvement/insist-that-devs-own-gihub-dir

Ensure that the cisagov devs are the owners of the .github directory

jsf9k committed:
Merge branch 'develop' into improvement/get-python-version-more-simply

jsf9k committed:
Merge pull request #59 from cisagov/improvement/get-python-version-more-simply

Use the python version output by actions/setup-python

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

Also updated the test and build jobs in the build workflow to mirror cache key
format changes in the lint job.

mcdonnnj committed:
Merge pull request #61 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into sort_requirements

mcdonnnj committed:
Merge pull request #59 from cisagov/sort_requirements

Sort the Package Requirements in setup.py

mcdonnnj committed:
Merge branch 'develop' into resolve_manual_check_disables

mcdonnnj committed:
Merge pull request #60 from cisagov/resolve_manual_check_disables

Remove or Explain Manual Testing Check Disables

mcdonnnj committed:
Clean up our actions/cache step

Removed name because it was not more informative than the default. Swapped out
a hardcoded job reference for the github.job context value. Switch the base
cache key to a step environment value so we can set it once and reuse. Removed
additional restore-key value that might have undesirable results.

mcdonnnj committed:
Update cache key used for test and build jobs

Use the full semver for the installed Python version in cache keys.

mcdonnnj committed:
Merge pull request #62 from cisagov/improvement/clean_up_actions/cache

Improve actions/cache Configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update test and build workflow jobs

Update the test and build workflow jobs to use the actions/cache style that was
done for the lint job.

mcdonnnj committed:
Merge pull request #63 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/fix_python_version_in_cache_keys

mcdonnnj committed:
Merge pull request #62 from cisagov/improvement/fix_python_version_in_cache_keys

Use Full semver of Installed Python Version in Cache Keys

mcdonnnj committed:
Update prettier hook details

Per https://github.com/prettier/prettier/pull/8937 the pre-commit hook has
been moved to https://github.com/pre-commit/mirrors-prettier. I have also
updated to the latest version in that repository.

mcdonnnj committed:
Update black hook repo url

At some point the python GitHub organization renamed to psf (Python Software
Foundation). Although it redirects with no issue, I think it should be updated
to reflect the current repository URL.

mcdonnnj committed:
Remove trailing '.git's from repository URLs

Two hooks had trailing '.git's in the URLs. Although this is not a problem, we
should be consistent in how we format things.

mcdonnnj committed:
Enable two additional hooks from default pre-commit-hooks

Enabled 'check-case-conflict' because of our mixed Linux and macOS development.
Although APFS supports case-sensitive containers, it is not the default as far
as I am aware. Linux filesystems are typically case-sensitive however.

With the merge of https://github.com/cisagov/development-guide/pull/42 we now
have a TOML file in a repository, so it does not hurt to add this hook in case
more are added in the future.

mcdonnnj committed:
Run pre-commit autoupdate

mcdonnnj committed:
Update repo URL for the isort hook

I missed that this repository was transfered from the creator, Timothy Crosley,
to the PyCQA organization.

mcdonnnj committed:
Update ansible-lint hook URL

mcdonnnj committed:
Run pre-commit autoupdate

mcdonnnj committed:
Add the pre-commit-packer repo and hooks

mcdonnnj committed:
Add the --strict flag to the yamllint pre-commot hook

The --strict flag will cause the yamllint hook to exit with a non-zero exit
code when warnings are found instead of only when errors are found.

mcdonnnj committed:
Merge pull request #63 from cisagov/pre-commit_hook_maintenance

pre-commit Hook Maintenance and Update

mcdonnnj committed:
Merge branch 'develop' into improvement/add_pre-commit-packer

mcdonnnj committed:
Merge pull request #64 from cisagov/improvement/add_pre-commit-packer

Add the pre-commit-packer Repo and Hooks to the pre-commit Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/enable_yamllint_strict

mcdonnnj committed:
Merge pull request #65 from cisagov/improvement/enable_yamllint_strict

Use the --strict Flag for the yamllint pre-commit Hook

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #64 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Add the beautysh pre-commit hook

This hook performs auto-formatting (beautifying) of Bash scripts.

mcdonnnj committed:
Update pre-commit hooks with `pre-commit autoupdate`

mcdonnnj committed:
Add the mypy package to dev rquirements

mcdonnnj committed:
Update the URL used for our Python packages

us-cert.gov now forwards to us-cert.cisa.gov, and since NCATS is a pre-CISA
designation it makes sense to update this URL. The URL points to a specific
set of services offered that does not encompass everything that is hosted in
the cisagov org on GitHub. We have chosen the page describing CISA's
cybersecurity mission as the best replacement URL.

mcdonnnj committed:
Update the author_email field in package metadata

We no longer have access to the old email address. Additionally the agency
has gotten it's own email domain, so we prefer cisa.dhs.gov addresses at
this point.

mcdonnnj committed:
Remove download_url field and add project_urls field to package metadata

The download_urls field was being used incorrectly, as it is supposed to
provide a download URL to a file usable to install the package. This would
typically be a tarball of some kind. Additionaly this field is discouraged
per https://github.com/pypa/packaging.python.org/issues/293, so we should
stop using it in our package metadata. The best replacement I found is the
project_urls field, which allows us to provide a number of purpose specific
URLs about the package.

mcdonnnj committed:
Fix capitalization of Python in description metadata

mcdonnnj committed:
Fix agency name in author metadata

mcdonnnj committed:
Revert ansible-lint version update

The v5 release of ansible-lint introduces breaking changes, so we are holding
off on updating until things are resolved. Conversation about this can be
tracked in https://github.com/cisagov/skeleton-ansible-role/issues/69.

mcdonnnj committed:
Merge pull request #69 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_beautysh_pre-commit_hook

mcdonnnj committed:
Merge pull request #67 from cisagov/improvement/add_beautysh_pre-commit_hook

Add the beautysh pre-commit hook

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Autoformat bump_version.sh with beautysh

mcdonnnj committed:
Merge pull request #68 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/add_mypy_to_dev_requirements

mcdonnnj committed:
Merge pull request #65 from cisagov/improvement/add_mypy_to_dev_requirements

Add the mypy Package to the Development Package Requirements

mcdonnnj committed:
Merge branch 'develop' into improvement/update_package_metadata

mcdonnnj committed:
Merge pull request #67 from cisagov/improvement/update_package_metadata

Update Package Metadata

mcdonnnj committed:
Update CodeQL workflow for GitHub Actions

I noticed the following warning when looking at GHA workflow runs:

1 issue was detected with this workflow: git checkout HEAD^2 is no longer
necessary. Please remove this step as Code Scanning recommends analyzing the
merge commit for best results.

After looking into it I updated the workflow per this page on GitHub Docs:

https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/troubleshooting-the-codeql-workflow#warning-git-checkout-head2-is-no-longer-necessary

mcdonnnj committed:
Merge pull request #69 from cisagov/bugfix/update_codeql_workflow

Update CodeQL Workflow for GitHub Actions

mcdonnnj committed:
Use secrets.GITHUB_TOKEN for coveralls in GHA workflow

coveralls-python has better integration with running in GitHub Actions now, so
we can use the automatically generated GITHUB_TOKEN secret instead of manually
setting a COVERALLS_REPO_TOKEN secret for each repository.

mcdonnnj committed:
Update coveralls GHA configuration for parallelization

We already ran our tests in parallel for each major version of Python a project
supports. This change should better differentiate each test job run in the
reports submitted.

mcdonnnj committed:
Skip CodeQL workflow for Dependabot branches on push events

This should resolve the following error:

Error: Workflows triggered by Dependabot on the "push" event run with read-only
access. Uploading Code Scanning results requires write access. To use Code
Scanning with Dependabot, please ensure you are using the "pull_request" event
for this workflow and avoid triggering on the "push" event for Dependabot
branches. See
https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-on-push
for more information on how to configure these events.

felddy committed:
Improve gitignore file comments and organization.

* Add a comment describing what the files does.
* Add sections to organize patterns.

hillaryj committed:
Adds a comment to direct additional requirements into setup.py

mcdonnnj committed:
Update pre-commit hooks

Hooks updated with the `pre-commit autoupdate` command.

Note:
The `ansible-lint` hook is intentionally being held back on 4.3.7 because of
ongoing issues with the 5.x version and how we use Ansible (standalone Galaxy
roles).

mcdonnnj committed:
Remove user from CODEOWNERS

Due to her departure this removes hillaryj from the default CODEOWNERS we use
in our projects.

hillaryj committed:
Merge pull request #74 from cisagov/add-req-comment

Adds a comment to direct additional requirements into setup.py

mcdonnnj committed:
Merge branch 'develop' into improvement/skip_codeql_on_push_for_dependabot_branches

mcdonnnj committed:
Merge pull request #73 from cisagov/improvement/skip_codeql_on_push_for_dependabot_branches

Skip CodeQL Workflow for Dependabot Branches on push Events

mcdonnnj committed:
Merge branch 'develop' into improvement/update_gha_coveralls_configuration

mcdonnnj committed:
Merge pull request #71 from cisagov/improvement/update_gha_coveralls_configuration

Update coveralls-python Configuration in GitHub Actions Workflow

mcdonnnj committed:
Update configuration instructions for pyenv

With the release of pyenv v2.0.0 there is a breaking change around startup
logic that necessitates an update for our pyenv setup instructions. Also
add a statement about how to get configuration instructions from pyenv
itself.

mcdonnnj committed:
Merge pull request #71 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into maintenance/remove_user_from_codeowners

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #72 from cisagov/maintenance/remove_user_from_codeowners

Remove User from CODEOWNERS

mcdonnnj committed:
Merge branch 'develop' into documentation/update_pyenv_instructions

mcdonnnj committed:
Merge pull request #75 from cisagov/documentation/update_pyenv_instructions

Update Instructions for Configuring pyenv

mcdonnnj committed:
Merge branch 'develop' into improvement/gitignore_me

mcdonnnj committed:
Merge pull request #70 from cisagov/improvement/gitignore_me

Improve gitignore file comments and organization.

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #76 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Standardize the layout of the Lineage configuration file

mcdonnnj committed:
Add to the cache keys for the GitHub Actions workflow

This adds a hash of `setup.py` to the cache keys used in the GitHub Actions
workflow for Python projects.

mcdonnnj committed:
Add comments explaining different hashFiles() argument

The other cache keys for our GHA jobs are in the format '**/<filename>' so that
any file with that name is used in the repository. However, for Python packages
they may have a 'setup.py' as part of their internal codebase that does not
impact environment requirements. As a result we only want to use the 'setup.py'
that is in the root of the repository and is used to install the package.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Add style enforcement rules

Add rules to enforce ATX-closed headers, dashes for unordered list elements,
and `1.` for ordered list elements.

mcdonnnj committed:
Add rule for image headers

Add <h1> and <img> tags to the allowed list for MD033 (HTML elements) to
support using an image as the first thing in a markdown file (header image).

mcdonnnj committed:
Switch to a YAML markdownlint configuration file

This converts the existing `.mdl_config.json` file to an equivalent
`.mdl_config.yaml` file. The reference in the markdownlint pre-commit hook
configuration is updated to match.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add comments to markdownlint configuration

Now that this is a YAML file we can add comments explaining the rule
modifications we use. This will make it easier to edit or expand in the future.

mcdonnnj committed:
Add the validate_manifest hook from pre-commit

This hook will validate any pre-commit hook manifest files in the repository.

jsf9k committed:
Add assertion to verify that the root logger's logging level is set correctly

See the discussion here for more context:
https://github.com/cisagov/pe-reports/pull/6#discussion_r657329612

jsf9k committed:
Install terraform and packer for the linting job

We should be doing this because the Packer and Terraform pre-commit
hooks leverage the corresponding executables; therefore, it makes
sense to go ahead and install the particular versions of those
executables that we support.  Also add support for optionally
debugging via tmate.

See also cisagov/skeleton-generic#74.

jsf9k committed:
Break out the curl cache creation into its own step

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove unnecessary line in tasks

There is no reason to create /usr/bin/terraform.  This is a vestige of
an earlier age.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Modify the Packer installation to model that of Terraform

The Terraform installation does not destroy the existing system
Terraform installation, and neither should the Packer installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Install a specific version of terraform-docs

Note that this change is dependent on the merging of
cisagov/setup-env-github-action#31.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Move go installation so that it takes place before the cache task

Some variables defined in the go installation are used in the cache
task, so the go installation must happen first.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Capitalize Go for consistency

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Update pre-commit hooks

This is performed by running `pre-commit autoupdate`, but with the
`ansible-lint` hook held back manually.

jsf9k committed:
Prefer the newer "go install" syntax

As of [Go 1.16](https://tip.golang.org/doc/go1.16#go-command) the `GO111MODULE` environment variable defaults to `on` and `go get` has been deprecated for module installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Install the shfmt tool for GHA

The `shfmt` tool does not ship on the GitHub Actions runners so we must install
it manually.

mcdonnnj committed:
Replace the beautysh hook with pre-commit-shfmt

We have had a difficult time with how beautysh parses some shellscripts. I went
in pursuit of an alternative and I believe shfmt to be a good alternative.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Apply changes from the shfmt pre-commit hook

mcdonnnj committed:
Remove unnecessary boilerplate

This package has a `__main__.py` file which removes the need to include this
boilerplate.

mcdonnnj committed:
Explicitly call sys.exit for errors

Instead of returning a value the main loop should explicitly call `sys.exit()`
if an error is encountered that would result in a non-zero exit code.

mcdonnnj committed:
Update function type hints

The type hints for the example_div function indicate it takes two float values,
but the command line interface indicates (and checks) that it takes integer
values.

mcdonnnj committed:
Do not use f-strings in logging messages

C-style '%' formatting should be used to any calls to the `logging` library.

mcdonnnj committed:
Update testing for exit code changes

The testing needs to be updated now that `example.main()` only returns a
`None` and error exits are done by directly calling `sys.exit()`.

mcdonnnj committed:
Merge pull request #83 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_pre-commit_hook

mcdonnnj committed:
Merge pull request #80 from cisagov/improvement/add_pre-commit_hook

Add the validate_manifest Hook to pre-commit Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/update_mdl_configuration

jsf9k committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #79 from cisagov/improvement/update_mdl_configuration

Update markdownlint Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #82 from cisagov/improvement/install-tf-and-packer-for-linting

Install terraform and packer for the linting job

mcdonnnj committed:
Merge branch 'develop' into improvement/replace_beautysh_with_shfmt

mcdonnnj committed:
Merge pull request #84 from cisagov/improvement/replace_beautysh_with_shfmt

Replace beautysh hook with cisagov/pre-commit-shfmt

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Changes from the pre-commit-shfmt hook

mcdonnnj committed:
Add stubs required by mypy pre-commit hook

With the release of 0.900 mypy moved to a modular stub model. This means that
only stdlib stubs are "baked in" and everything else must be installed
separately. Since the mypy hook runs in a venv the stubs it recommends need to
be installed as additional dependencies.

jsf9k committed:
Install mypy type stubs as part of setup-env

We already do this as part of the mypy pre-commit hook, but this way
mypy is also ready to run manually in the developer's local Python
virtual env.

mcdonnnj committed:
Merge pull request #82 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/standardize_lineage_configuration

mcdonnnj committed:
Merge pull request #77 from cisagov/improvement/standardize_lineage_configuration

Standardize the Layout of the Lineage Configuration File

mcdonnnj committed:
Merge branch 'develop' into improvement/improve-test

mcdonnnj committed:
Merge pull request #80 from cisagov/improvement/improve-test

Verify that the logging level for the root logger is set correctly

mcdonnnj committed:
Merge branch 'develop' into improvement/changes_to_match_development_preferences

mcdonnnj committed:
Merge pull request #81 from cisagov/improvement/changes_to_match_development_preferences

Update to Match Best Practices

mcdonnnj committed:
Merge branch 'develop' into improvement/add_setup_to_gha_cache_key

mcdonnnj committed:
Merge pull request #78 from cisagov/improvement/add_setup_to_gha_cache_key

Add setup.py to the Cache Keys for the GitHub Actions Workflow

mcdonnnj committed:
Switch to block-style sequence in build workflow

Switch from a flow style to a block style sequence in the build workflow. This
will improve maintainability by making version changes impact one Python
version at a time.

dav3r committed:
Fix a typo

mcdonnnj committed:
Use the hashicorp/setup-terraform Action

Instead of manually installing a Terraform binary we can leverage the Action
provided by Hashicorp to do the same thing.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Add a markdownlint rule for horizontal rules

This sets a specific rule for MD035 (Horizontal rule style) instead of the
default value of "consistent".

mcdonnnj committed:
Add a markdownlint rule for code blocks

This sets a specific rule for MD046 (code block style) instead of the default
value of "consistent".

mcdonnnj committed:
Merge branch 'develop' into dav3r-patch-1

mcdonnnj committed:
Merge pull request #85 from cisagov/dav3r-patch-1

Fix a typo

mcdonnnj committed:
Merge branch 'develop' into improvement/specify_mdl_rules

mcdonnnj committed:
Merge pull request #89 from cisagov/improvement/specify_mdl_rules

Add markdownlint Rules to Enforce Specific Formatting

mcdonnnj committed:
Merge branch 'develop' into improvement/use_setup-terraform_in_gha_workflow

mcdonnnj committed:
Merge pull request #88 from cisagov/improvement/use_setup-terraform_in_gha_workflow

Use the hashicorp/setup-terraform Action to Install Terraform in GHA Workflow

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

dav3r committed:
Enable terraform validate pre-commit hook

This hook should now work as expected in most, if not all of our repos 
now that we are finally updating to terraform 0.13.x (on our way to 
1.0.x).

mcdonnnj committed:
Merge pull request #85 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/gha_python_version_block_style_sequence

mcdonnnj committed:
Merge pull request #83 from cisagov/improvement/gha_python_version_block_style_sequence

Switch to Block-Style Sequence for Python Versions in build Workflow

dav3r committed:
Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13

This change will be reverted when testing is completed.

dav3r committed:
Remove lint job step to initialize Terraform directories

Initialization will now be done during the "terraform validate" step.

dav3r committed:
Revert "Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13"

This reverts commit 6a7fbf07bd371d0493c523ce24647e5c04c77c03.

dav3r committed:
Merge pull request #90 from cisagov/improvement/enable_terraform_validate

Enable terraform validate pre-commit hook

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Use an id when using cisagov/setup-env-github-action

mcdonnnj committed:
Use setup-env outputs for Terraform version

Change from using an environment variable to the outputs of the
cisagov/setup-env-github-action instead.

mcdonnnj committed:
Use setup-env outputs for Packer version

We use a `PACKER_VERSION` environment variable for the "Install Packer" step
that is populated from the `setup-env` outputs to get around `yamllint` lint
length limits.

mcdonnnj committed:
Use setup-env outputs for shfmt version

We change the "Install shfmt" step to use two local environment variables to
provide the package's URL and version to install. This allows us to work around
`yamllint` line length limits.

mcdonnnj committed:
Use setup-env outputs for Terraform-docs version

We change the "Install Terraform-docs" step to use two local environment
variables to provide the package's URL and version to install. This allows us
to work around `yamllint` line length limits.

mcdonnnj committed:
Store Go version as a step output

This changes from using an environment variable to using a step output to store
the Go version that is installed. This mirrors changes made to the other
program versions and how they're stored.

mcdonnnj committed:
Update the ansible-lint version for pre-commit

mcdonnnj committed:
Add an ansible-lint configuration file

This file is copied from cisagov/skeleton-ansible-role#85 and tweaked to pass
pre-commit. See here for for a list of the elements that can exist in this
file: https://ansible-lint.readthedocs.io/en/latest/configuring.html

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add Python 3.10 to GHA version matrices

Add Python 3.10 to the matrices of versions used by the `test` and `build`
steps.

mcdonnnj committed:
Force strings for GHA version matrices

Ensure that the versions listed in the version matrices used by the `test` and
`build` jobs are interpreted as strings. Without this change the version `3.10`
is interpreted as `3.1` when converted to a fixnum. Since these are intended to
be interpreted as string values we should enforce it for all versions listed.

mcdonnnj committed:
Add dependabot configuration for Terraform

This will configure `dependabot` to scan Terraform configurations if they exist
in a repository. If a repository stores a Terraform configuration in a sub-
directory this configuration will need to be modified or an additional
configuration must be added if there still exists a Terraform configuration in
the root directory.

mcdonnnj committed:
Merge pull request #91 from cisagov/improvement/pre-commit_update

Update `pre-commit` Hooks

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/update_ansible-lint_pre-commit_version

Update the `ansible-lint` Version in the pre-commit Configuration

mcdonnnj committed:
Merge pull request #94 from cisagov/improvement/add_dependabot_configuration_for_terraform

Add `dependabot` Configuration for Terraform

mcdonnnj committed:
Merge pull request #92 from cisagov/improvement/use_action_outputs

Use Outputs of cisagov/setup-env-github-action to Control Installed Program Versions

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Update ansible-lint configuration

Remove the manual exclusion of the `converge.yml` file in the default molecule
scenario. With the changes in cisagov/skeleton-ansible-role#88 this exclusion
is no longer necessary.

mcdonnnj committed:
Update ansible-lint configuration

Instead of excluding molecule configurations for systemd individually we
instead direct ansible-lint to process any of these configurations in
scenario directories as plain yaml files. This mirrors how ansible-lint
handles `molecule.yml` files in scenario directories by default.

mcdonnnj committed:
Merge pull request #86 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #87 from cisagov/improvement/add_support_for_py3.10

Add Support for Python 3.10

mcdonnnj committed:
Disable fail-fast for matrix GHA jobs

Override the default by setting `fail-fast` to `false` for the test and build
jobs in our GHA build workflow. This ensure that all matrix jobs run regardless
of a failure.

mcdonnnj committed:
Merge pull request #88 from cisagov/improvement/disable_fail-fast_for_test_and_build_jobs

Disable `fail-fast` for GitHub Actions Jobs with a Matrix Strategy

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Merge pull request #97 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/update_ansible-lint_configuration

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_ansible-lint_configuration

Update `ansible-lint` Configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #89 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Add Python 3.10 to package classifiers

This was missed when we added testing/building for Python 3.10.

mcdonnnj committed:
Bump version from 0.0.1 to 0.1.0

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Add comment to Dependabot configuration

This comment explains that the configuration may have commented out
ignore directives that should be uncommented in downstream projects.

mcdonnnj committed:
Disable comments-indentation rule for yamllint

yamllint does not like it when you comment out pieces of dictionaries
in lists. Upcoming additions to the Dependabot configuration will run
afoul of this so we are updating the yamllint configuration.

mcdonnnj committed:
Merge pull request #99 from cisagov/improvement/update_dependabot_and_yamllint_configurations

Add boilerplate for upcoming Dependabot configuration changes

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #98 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Use asterisks for emphasis- and strong-styles

Given the inconsistent behavior of underscores used within words for
style we should prefer asterisks now that these rules are available.

mcdonnnj committed:
Add missing hyphen in markdownlint rule header

mcdonnnj committed:
Merge pull request #100 from cisagov/improvement/update_markdownlint_configuration

Use new `markdownlint` rules for emphasis- and strong-styles

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

dependabot[bot] committed:
Bump actions/setup-python from 2 to 3

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 2 to 3.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #91 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/add_official_py3.10_support

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/add_official_py3.10_support

Officially support Python 3.10

mcdonnnj committed:
Update git ignore list

Add dist/ to the .gitignore file to better support local building.

mcdonnnj committed:
Modernize the Python package build step

Modernize the build process by using the build package instead of the
now deprecated setup.py install method. Remove installing the package
requirements as it is unnecessary for this build process. Upgrade the
setuptools package as part of the initial dependency installation.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Add Dependabot ignore directives

This adds commented out ignore directives for the following GitHub
Actions:

- action/cache
- action/checkout
- action/setup-python

These should be uncommented downstream to ensure that updates to these
dependencies are pushed from pull requests made in the skeleton.

mcdonnnj committed:
Upgrade from Python 3.9 to 3.10 for the `lint` job in GHA

mcdonnnj committed:
Use consistent quoting for software versions

mcdonnnj committed:
Merge github.com:cisagov/skeleton-python-library into improvement/update_from_skeleton

mcdonnnj committed:
Apply changes from the `prettier` pre-commit hook

mcdonnnj committed:
Update to pass the `ansible-lint` pre-commit hook

mcdonnnj committed:
Update to pass the `isort` pre-commit hook

mcdonnnj committed:
Update the configure of the `mypy` pre-commit hook

mcdonnnj committed:
Move the `docker-compose` requirement

The `docker-compose` package is not required by the Python package but
as part of the environment for operation. It has been moved to
`requirements.txt` to reflect where it is needed.

mcdonnnj committed:
Merge pull request #101 from cisagov/dependabot/github_actions/actions/setup-python-3

Bump actions/setup-python from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into dependabot/github_actions/actions/checkout-3

mcdonnnj committed:
Merge pull request #102 from cisagov/dependabot/github_actions/actions/checkout-3

Bump actions/checkout from 2 to 3

dependabot[bot] committed:
Bump actions/cache from 2 to 3

Bumps [actions/cache](https://github.com/actions/cache) from 2 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge branch 'develop' into improvement/add_dependabot_ignores_for_github_actions

mcdonnnj committed:
Merge pull request #107 from cisagov/improvement/add_dependabot_ignores_for_github_actions

Add Dependabot ignore directives

mcdonnnj committed:
Merge pull request #104 from cisagov/dependabot/github_actions/actions/cache-3

Bump actions/cache from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into improvement/use_python_3.10_for_gha

mcdonnnj committed:
Merge pull request #106 from cisagov/improvement/use_python_3.10_for_gha

Use Python 3.10 for the `lint` job in GitHub Actions

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #105 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Enable Dependabot ignore directives

This enables the Dependabot ignore directives for GitHub Actions.

mcdonnnj committed:
Add permission scoping for the CodeQL Analyze GHA job

This sets permissions around `GITHUB_TOKEN` to the minimum necessary
for this workflow. Please see
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
for more information.

mcdonnnj committed:
Update github/codeql-action/init from 1 to 2

Update the comments to better mirror the current documentation for
github/codeql-action. Switch to block style for the map of languages.

mcdonnnj committed:
Consistently use block style in the CodeQL GHA workflow

mcdonnnj committed:
Update github/codeql-action/autobuild from 1 to 2

Clean up the commenting while we're here.

mcdonnnj committed:
Update github/codeql-action/analyze from 1 to 2

Also clean up the commenting while we're here.

mcdonnnj committed:
Add pre-commit hooks to validate GitHub Actions content

Add hooks to validate GitHub Actions and GitHub Action workflows. These
are validated using JSON Schemas.

mcdonnnj committed:
Merge pull request #109 from cisagov/improvement/add_github_actions_pre-commit_hooks

Add `pre-commit` hooks to validate GitHub Actions content

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Disable tests that require running containers

Disable any of the tests that would require the container orchestration
defined in the Docker Compose for this project. It would take some work
to support this in our current esting workflow and that work can be put
off until a later time.

mcdonnnj committed:
Use List from typing library

Subscripting base types is not supported until Python 3.9. We can fall
back on the typing library to ensure we support older Python versions.

mcdonnnj committed:
Merge pull request #97 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #98 from cisagov/improvement/update_codeql_workflow

Update the CodeQL workflow

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_build_process

Modernize the `build` step in the `build` workflow

mcdonnnj committed:
Merge github.com:cisagov/skeleton-python-library into improvement/update_from_skeleton

dav3r reviewed (approved):
This all looks good to me, though maybe we should add an issue here to get the commented-out tests (the ones that require the Docker composition to start up) running again or else come up with other tests that can take their place.

jsf9k commented on tests/cert_test.py:
```suggestion
# This test currently requires the Docker Compose configuration to be running
```

jsf9k reviewed (approved):
Other than one tiny grammar thang in a comment, this all looks great to me.  Thanks for bringing this one into the 21st century!

mcdonnnj committed:
Fix the grammar in a test comment

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj commented:
> This all looks good to me, though maybe we should add an issue here to get the commented-out tests (the ones that require the Docker composition to start up) running again or else come up with other tests that can take their place.

@dav3r Please see #9 for tracking this super-duper high priority task 😏

dav3r commented:
> > This all looks good to me, though maybe we should add an issue here to get the commented-out tests (the ones that require the Docker composition to start up) running again or else come up with other tests that can take their place.
> 
> @dav3r Please see #9 for tracking this super-duper high priority task 😏

Thanks @mcdonnnj for writing up that issue! 👍

mcdonnnj committed:
Update Dockerfile configuration

Update the Dockerfile to use Python 3.10.4 from 3.8. Update to include
improvements to how we create Dockerfiles from cisagov/skeleton-docker.

jsf9k commented:
@mcdonnnj, is this one ready to merge yet?  #5 needs these changes.

mcdonnnj commented:
> @mcdonnnj, is this one ready to merge yet? #5 needs these changes.

@jsf9k I'm trying to coordinate with @king-alexander to test his code rebased on this to make sure nothing's been broken by me.

mcdonnnj commented:
Re-requesting reviews because of the changes to `Dockerfile-admiral`.

jsf9k reviewed (approved):
_Approval intensifies_

dav3r commented on tests/cert_test.py:
```suggestion
# See https://github.com/cisagov/admiral/issues/9 for details.
# class TestCerts:
```

dav3r commented on tests/scan_test.py:
```suggestion
# See https://github.com/cisagov/admiral/issues/9 for details.
# class TestPortScans:
```

dav3r reviewed (approved):
👍 👍  Though I suggested inserting a couple of links to https://github.com/cisagov/admiral/issues/9 in comments.

mcdonnnj committed:
Add link to issue about fixing commented out tests

Improve the comments explaining the commented out tests by providing a
link to the issue created to track fixing them.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
